src/snowflake/snowpark/column.py

```diff
@@ -591,11 +699,18 @@ def regexp(self, pattern: ColumnOrLiteralStr) -> "Column":
         :meth:`rlike` is an alias of :meth:`regexp`.
 
         """
+        # TODO: Should SpColumnRegexp.pattern be an Expr in the IR, could sbe SpColumnExpr as in SpColumnLike?
```

Comment:
There seem to be a few TODOs which require IR changes, maybe collect in doc, and then go through in meeting? We prob. can resolve those quickly.